### ISSUE TITLE
About model.config.use_cache during training

### ISSUE THREAD

**Issue by Ziha0-Zhang** [jiangxinke/Agentic-RAG-R1#5]
In Agentic-RAG-R1/src/utils/utils.py:
```
def optimize_model_memory(model):
    """Apply memory optimizations like proper gradient checkpointing setup"""
    # Ensure model is in training mode
    model.train()

    # Disable caching for gradient checkpointing
    # model.config.use_cache = False
    model.config.use_cache = True

    # Enable gradient checkpointing
    model.gradient_checkpointing_enable()

    # Enable input gradients properly
    if hasattr(model, "enable_input_require_grads"):
        model.enable_input_require_grads()
    else:

        def make_inputs_require_grad(module, input, output):
            output.requires_grad_(True)

        model.get_input_embeddings().register_forward_hook(make_inputs_require_grad)

    return model
```
I‘m confused about `model.config.use_cache = True`

During training, it is common to disable caching (i.e., `use_cache = False`) during gradient checkpointing in order to save GPU memory. In this mode, intermediate activation values are recomputed during the backward pass rather than being stored during the forward pass. However, in this piece of code, `use_cache` is set to True, which means that the intermediate activation values will be cached instead of being recomputed.

**Comment by jiangxinke** [comment]
Close this issue as this problem has been solved
